src/test/fuzz/netbase_dns_lookup.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <netbase.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+namespace {
+FuzzedDataProvider* fuzzed_data_provider_ptr = nullptr;
+
+std::vector<CNetAddr> fuzzed_dns_lookup_function(const std::string& name, bool allow_lookup)
+{
+    std::vector<CNetAddr> resolved_addresses;
+    while (fuzzed_data_provider_ptr->ConsumeBool()) {
+        resolved_addresses.push_back(ConsumeNetAddr(*fuzzed_data_provider_ptr));
```

Comment:
After reading the `getaddrinfo` manpage, I realized `ConsumeNetAddr` can return `CNetAddr` with `m_scope_id` set for non link-local addresses, which is different from `getaddrinfo` behaviour. I'm not an expert here but I think there could be sanity-checking for that in `ConsumeNetAddr`.

Reply:
I don't think that is likely to matter in practice (don't hesitate to prove me wrong!). Leaving unchanged for now.